io-engine/src/rebuild/rebuild_task.rs

```diff
@@ -124,8 +127,14 @@ impl RebuildTask {
             &mut copy_buffer
         };
 
-        source_hdl.set_read_mode(ReadMode::UnwrittenFail);
+        // Check if the block has to be copied.
+        if let Some(ref log) = self.rebuild_log {
```

Comment:
So seems like we still go through entire range and simply skip non-modified segments.
Would it be better to simply "walk the log" ?

Reply:
My intention was to preserve the original algorithm, so we can say that it is valid. Just one "if" with obvious behavior won't brake it.

I don't see any performance issues here. The entire process is heavily I/O bound.

If I had to rewrite the logic to walk the log, we would end up with entire new code, with no real-world performance gain.

Reply:
I take your point, but couldn't we simply change `send_segment_task` to walk the log instead of walking the range? This wouldn't be a change of the algorithm, but perhaps I'm oversimplifying it :)

Reply:
One thing I guess we need to ensure is that we don't "walk the log" without locking the lba range, which is handled by plugging into existing logic.

Reply:
that is handled by each copy task, so that would not have to change when "walking the log".

Reply:
We can address the way tasks are walked in as a separate task, if the current implementation (walk the range) turns out to have performance problems.